airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/destination.py

```diff
@@ -69,7 +69,8 @@ def write(
                     writer.queue_write_operation(record.stream)
             elif message.type == Type.STATE:
                 # yielding a state message indicates that all preceding records have been persisted to the destination
-                writer.write_whats_left()
+                # cleaning records_buffer for target stream.
+                writer.clear_buffer(record.stream)
```

Comment:
@bazarnov Won't this change have the effect of erasing and not sending any records in the queues when we get a state message?  

Assuming we get messages like this:

```
RECORD
RECORD
STATE
```

We'll have 2 records in the buffer that haven't been set yet (didn't hit the 500 message threshold).  We need to `write_from_queue` before `clear_buffer`, no?

Reply:
Ah, my bad. You should be right, just because this is the `destination` not `source` and we'are working with the `source` messages. Reverted this piece.